File: ZipDownloadServer/Acceptance of Delivery.go
type AcceptOrder struct {
	Buyer string `json:"buyer"`
	Shipper string `json:"shipper"`
	AgreedDeliveryDate string `json:"agreedDeliveryDate"`
	OrderAccepted bool `json:"orderAccepted"`
	Type string `json:"type"`	
}
func (s *SmartContract) AcceptOrder(ctx contractapi.TransactionContextInterface, reqOrderId string, acceptOrderId string, buyer string, shipper string,
	agreedDeliveryDate string,orderAccepted string) error {

		orderAsBytes, err := ctx.GetStub().GetState(reqOrderId)
	if err != nil {
		return fmt.Errorf("Failed to read from world state. %s", err.Error())
	}

	if orderAsBytes == nil {
		return fmt.Errorf("%s does not exist", reqOrderId)
	}
	details := new(RequestOrder)
	_ = json.Unmarshal(orderAsBytes, details)
	orderAccepted1, _ := strconv.ParseBool(orderAccepted)
	details.OrderAccepted = orderAccepted1;
	details.AgreedDeliveryDate=agreedDeliveryDate;
    Type:="acceptorder"

	detailsAsBytes, _ := json.Marshal(details)
	ctx.GetStub().PutState(reqOrderId, detailsAsBytes)
      	

		details1 :=AcceptOrder{
		Buyer:buyer,
		Shipper:shipper,
		AgreedDeliveryDate:agreedDeliveryDate,
		OrderAccepted:orderAccepted1,
		Type:Type,
	}


	detailsAsBytes1, _ := json.Marshal(details1)

	return ctx.GetStub().PutState(acceptOrderId, detailsAsBytes1)
}

File: ZipDownloadServer/Acceptance of Delivery.js
async  AcceptanceOfDelivery(ctx,acceptanceofDeliveryId,reqOrderId,actualDeliveryDate1){
    let status="";
    const orderAsBytes = await ctx.stub.getState(reqOrderId); 
        if (!orderAsBytes || orderAsBytes.length === 0) {
            throw new Error(`${reqOrderId} does not exist`);
        }
        const order = JSON.parse(orderAsBytes.toString());
        let agreedDeliveryDate1=order.agreedDeliveryDate;
        let terminationDays1=order.terminationDays;
        //lateFee1=order.lateFee;
        let shipper1=order.shipper;
        let buyer1=order.buyer;
        order.actualDeliveryDate=actualDeliveryDate1;
        await ctx.stub.putState(reqOrderId, Buffer.from(JSON.stringify(order)));

    const date1 = new Date(actualDeliveryDate1);
    const date2 = new Date(agreedDeliveryDate1);
    if(date2>date1){
        status="PASSED_TESTING";
        console.log("PASSED_TESTING");
    }
    const diffTime = Math.abs(date2 - date1);
    const diffDays = Math.ceil(diffTime / (1000 * 60 * 60 * 24)); 
    if(diffDays>terminationDays1)
    {
        status="OUTSIDE_INSPECTION_PERIOD";
        console.log("OUTSIDE_INSPECTION_PERIOD");
    }else{
        status="PASSED_TESTING";
        console.log("PASSED_TESTING");
    }
    let type="acceptanceofdelivery";
    const details = {
        "acceptanceofDeliveryId":acceptanceofDeliveryId,
        "shipper":shipper1,
        "buyer":buyer1,
        "actualDeliveryDate":actualDeliveryDate1,
        "agreedDeliveryDate":agreedDeliveryDate1,
        "terminationDays":terminationDays1,
        "status":status,
        "diffDays":diffDays,
        "type":type,
    };
    await ctx.stub.putState(acceptanceofDeliveryId, Buffer.from(JSON.stringify(details)));

    return details;
}
